operator_tracker/src/main.rs

```diff
@@ -0,0 +1,252 @@
+#[derive(argh::FromArgs)]
+struct Args {
+    /// path to the .env file
+    #[argh(option)]
+    env_file: Option<String>,
+}
+
+#[tokio::main]
+async fn main() {
+    let args: Args = argh::from_env();
+
+    if let Some(env_file) = args.env_file {
+        dotenv::from_filename(env_file).expect("Failed to load env file");
+    } else {
+        dotenv::dotenv().expect("Failed to load env file");
+    }
+
+    env_logger::init();
+
+    let database_url = &std::env::var("DATABASE_URL").expect("DATABASE_URL must be set");
+
+    let rpc_url = &std::env::var("RPC_URL").expect("RPC_URL must be set");
+
+    let registry_coordinator_addr = &std::env::var("REGISTRY_COORDINATOR_ADDRESS")
+        .expect("REGISTRY_COORDINATOR_ADDRESS must be set");
+
+    let registry_coordinator_addr: Address = registry_coordinator_addr
+        .parse()
+        .expect("Failed to parse registry coordinator addr");
+
+    let pool = PgPoolOptions::new()
+        .max_connections(5)
+        .connect(database_url)
+        .await
+        .expect("Failed to create connection pool");
+
+    pool.execute(include_str!("../sql/schema.sql"))
+        .await
+        .expect("Failed to initialize DB");
+
+    let eth_rpc = Arc::new(Provider::<Http>::connect(rpc_url).await);
+
+    let registry_coordinator = RegistryCoordinator::new(registry_coordinator_addr, eth_rpc.clone());
+
+    let state = AppState {
+        pool,
+        registry_coordinator,
+    };
+
+    let listener = tokio::net::TcpListener::bind("0.0.0.0:3000")
+        .await
+        .expect("Failed to bind listener");
+
+    let router = Router::new()
+        .route("/versions", post(post_operator_version))
+        .route("/versions", get(list_operator_versions))
+        .with_state(state);
+
+    axum::serve(listener, router)
+        .await
+        .expect("Failed to serve app");
+}
+
+#[derive(serde::Deserialize)]
+struct OperatorVersionPayload {
+    pub version: String,
+    pub signature: String,
+}
+
+#[derive(serde::Serialize, Debug)]
+enum OperatorVersionError {
+    InvalidSignature,
+    OperatorNotRegistered,
+}
+
+impl IntoResponse for OperatorVersionError {
+    fn into_response(self) -> axum::http::Response<axum::body::Body> {
+        axum::http::Response::builder()
+            .status(axum::http::StatusCode::BAD_REQUEST)
+            .body(axum::body::Body::from(
+                serde_json::to_string(&self).unwrap(),
+            ))
+            .unwrap()
+    }
+}
+
+#[derive(sqlx::FromRow, Debug, Serialize)]
+struct OperatorVersion {
+    address: String,
+    version: String,
+}
+
+impl IntoResponse for OperatorVersion {
+    fn into_response(self) -> axum::http::Response<axum::body::Body> {
+        axum::http::Response::builder()
+            .status(axum::http::StatusCode::OK)
+            .body(axum::body::Body::from(
+                serde_json::to_string(&self).unwrap(),
+            ))
+            .unwrap()
+    }
+}
+
+async fn post_operator_version(
```

Comment:
This should return 201 if it is created correctly. Also should include a link to the entity in the `Location` header and the created entity as response body